tests/test_hdf5.py

```diff
@@ -56,6 +56,21 @@ def test_h5py_dataset_split_parsing():
     os.remove('tmp.hdf5')
 
 
+def test_h5py_dataset_axis_labels():
+    h5file = h5py.File(name='tmp.hdf5', mode="w")
```

Comment:
@vdumoulin Could you wrap this into a `try-finally` block? Right now `tmp.hdf5` won't be removed if the test fails for any reason.